spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -146,13 +148,31 @@ public RewriteDataFilesSparkAction filter(Expression expression) {
     return this;
   }
 
+  @Override
+  public RewriteDataFiles targetBranch(String branch) {
+    this.targetBranch = branch;
+    SnapshotRef ref = table.refs().get(this.targetBranch);
```

Comment:
what if the user passes the tag name which exists? Should we throw an exception by catching this case early?

Reply:
good point, I think it should be prohibited.